storm-core/src/jvm/backtype/storm/topology/WindowedBoltExecutor.java

```diff
@@ -0,0 +1,200 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package backtype.storm.topology;
+
+import backtype.storm.Config;
+import backtype.storm.task.IOutputCollector;
+import backtype.storm.task.OutputCollector;
+import backtype.storm.task.TopologyContext;
+import backtype.storm.tuple.Tuple;
+import backtype.storm.windowing.TupleWindowImpl;
+import backtype.storm.windowing.WindowLifecycleListener;
+import backtype.storm.windowing.WindowManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static backtype.storm.topology.base.BaseWindowedBolt.Count;
+import static backtype.storm.topology.base.BaseWindowedBolt.Duration;
+
+/**
+ * An {@link IWindowedBolt} wrapper that does the windowing of tuples.
+ */
+public class WindowedBoltExecutor implements IRichBolt {
+    private static final Logger LOG = LoggerFactory.getLogger(WindowedBoltExecutor.class);
+
+    private IWindowedBolt bolt;
+    private transient WindowedOutputCollector windowedOutputCollector;
+    private transient WindowLifecycleListener<Tuple> listener;
+    private transient WindowManager<Tuple> windowManager;
+
+    public WindowedBoltExecutor(IWindowedBolt bolt) {
+        this.bolt = bolt;
+    }
+
+    private int getTopologyTimeoutMillis(Map stormConf) {
+        if (stormConf.containsKey(Config.TOPOLOGY_ENABLE_MESSAGE_TIMEOUTS)) {
+            boolean timeOutsEnabled = (boolean) stormConf.get(Config.TOPOLOGY_ENABLE_MESSAGE_TIMEOUTS);
+            if (!timeOutsEnabled) {
+                return Integer.MAX_VALUE;
+            }
+        }
+        int timeout = 0;
+        if (stormConf.containsKey(Config.TOPOLOGY_MESSAGE_TIMEOUT_SECS)) {
+            timeout = ((Number) stormConf.get(Config.TOPOLOGY_MESSAGE_TIMEOUT_SECS)).intValue();
+        }
+        return timeout * 1000;
+    }
+
+    private void ensureDurationLessThanTimeout(int duration, int timeout) {
+        if (duration > timeout) {
+            throw new IllegalArgumentException("Window duration (length + sliding interval) value " + duration +
+                                                       " is more than " + Config.TOPOLOGY_MESSAGE_TIMEOUT_SECS +
+                                                       " value " + timeout);
+        }
+    }
+
+    // TODO: add more validation
```

Comment:
If we need more validation we should add it in.  If we don't we should remove this comment.